addon/components/frost-list-item-content.js

```diff
@@ -0,0 +1,103 @@
+import Ember from 'ember'
+const {get, isPresent} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import {Component} from 'ember-frost-core'
+import {PropTypes} from 'ember-prop-types'
+
+import layout from '../templates/components/frost-list-item-content'
+
+export default Component.extend({
+
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+  layout,
+
+  // == PropTypes =============================================================
+
+  propTypes: {
+    // Options - required
+    model: PropTypes.EmberObject.isRequired,
+    index: PropTypes.number.isRequired,
+    onExpand: PropTypes.func.isRequired,
+    onSelect: PropTypes.func.isRequired,
+
+    // Options - general
+    onSelectionChange: PropTypes.func,
+    itemTypes: PropTypes.object,
+    itemTypeKey: PropTypes.string,
+    isAnyTypedItemExpansion: PropTypes.bool,
+
+    // Options - sub-components
+    item: PropTypes.EmberComponent.isRequired,
+    itemExpansion: PropTypes.EmberComponent
+  },
+
+  getDefaultProps () {
+    return {
+      isAnyTypedItemExpansion: false
+    }
+  },
+
+  // == Computed Properties ===================================================
+
+  @readOnly
+  @computed('model', 'itemTypes', 'itemTypeKey')
+  typedItemComponent (model, itemTypes, itemTypeKey) {
+    if (isPresent(model) && isPresent(itemTypes) && isPresent(itemTypeKey)) {
```

Comment:
What will happen if itemTypes is not defined? I'm expecting that that `item` (passed in parameter) will be returned.

Reply:
If `itemTypes` is not defined, this computed property will be undefined, but I have a another computed property lower down in the file that depends on this one, and returns either the resulting component of this computed property, or the `item` component that's passed in.